File: TaskManagementApi/Core/TaskManagement.Domain/Models/TaskModel.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManagement.Domain.Models
{
    public class TaskModel
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }
        public DateTime CreateDate { get; set; } = DateTime.Now;
        public DateTime DueDate { get; set; }

        public bool Completed { get; set; }
        
        public int UserId { get; set; }
    }
}

File: TaskManagementWeb/src/app/shared/models/TaskModel.ts
export class TaskModel {
  Id!: number;
  Name: string;
  Description: string;
  DueDate: Date;
  CreateDate: Date;
  Completed: boolean;
  UserId: number
  constructor(
  Id: number,
  Name: string,
  Description: string,
  DueDate: Date,
  CreateDate: Date,
  Completed: boolean,
  UserId: number
  ) {
    this.Id = Id;
    this.Name = Name;
    this.Description = Description;
    this.DueDate = DueDate;
    this.CreateDate = CreateDate;
    this.Completed = Completed;
    this.UserId = UserId;
  }
}
